app/Domain/Achievements/Models/Achievement.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace App\Domain\Achievements\Models;
+
+use App\Domain\Achievements\Enums\AchievementType;
+use App\Domain\Experience\Enums\ExperienceType;
+use App\Domain\Experience\Projections\UserAchievementProjection;
+use App\Models\Series;
+use Database\Factories\AchievementFactory;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class Achievement extends Model
+{
+    use HasFactory;
+
+    protected $casts = [
+        'type' => AchievementType::class,
+        'data' => 'array',
+    ];
+
+    public function scopeForSeries(Builder $builder, Series $series): void
+    {
+        $builder->whereJsonContains('data->series_id', $series->id);
+    }
+
+    public function scopeForPullRequest(Builder|Achievement $builder): void
+    {
+        $builder->where('type', AchievementType::PullRequest());
+    }
+
+    public function scopeForExperience(Builder|Achievement $builder): void
+    {
+        $builder->where('type', AchievementType::Experience());
+    }
+
+    public function receivedBy(int $userId): bool
+    {
+        return UserAchievementProjection::forUser($userId)
+            ->andSlug($this->slug)
+            ->exists();
+    }
+
+    protected static function newFactory()
+    {
+        return new AchievementFactory();
+    }
```

Comment:
This is not really required unless namespace is other than `Database\Factories` 🤔

Reply:
Laravel crashed without this one though… maybe because the model itself is in a different namespace?

Reply:
I thought it's only applies to Factories namespace, I haven't tested though https://laravel.com/docs/8.x/database-testing#connecting-factories-and-models

Reply:
> I thought it's only applies to Factories namespace, I haven't tested though https://laravel.com/docs/8.x/database-testing#connecting-factories-and-models

It's required for models that are out of the default namespace. By default Laravel strips "Factory" from the end of the factory class name and attempts to resolve the model in either the app directory or app models directory.

This is why Laravel has the `newFactory()` method on the `HasFactory` trait.

See here for reference: https://github.com/laravel/framework/blob/a60b498301cf2edb3f379246565b850080fe2a82/src/Illuminate/Database/Eloquent/Factories/Factory.php#L636-L649